src/reducers/board.js

```diff
@@ -1,10 +1,27 @@
-import {ActionTypes, BoardConfigurations} from '../constants';
+import {ActionTypes, BoardConfigurations, CellStates} from '../constants';
 import {initializeBoard} from './helpers/board';
 
-const {INITIAL_STATE} = ActionTypes;
+const {INITIAL_STATE, BOARD_TILE_SELECTED} = ActionTypes;
 
 function board(state = [], action) {
   switch (action.type) {
+    case BOARD_TILE_SELECTED:
+      if (action.rack !== '' && state[action.selected.row][action.selected.col].status === CellStates.EMPTY) {
```

Comment:
one tip to writing cleaner code is to substitute expressions with functions:

```
if (isValidSelection(state, action)) {}
```

or 

```
if (doesActionHaveRack(action) && isBoardCellEmpty(state, action)) {}
```